packages/core/src/testers/index.ts

```diff
@@ -345,17 +345,32 @@ export const isObjectArrayWithNesting =
     }
     const schemaPath = (uischema as ControlElement).scope;
     const resolvedSchema = resolveSchema(schema, schemaPath);
+    const minimalElementsMap: { [key: string]: number; } = { 'object': 1, 'array': 0 };
```

Comment:
The name of `minimalElementsMap` is a bit misleading. Maybe rename this to `minNestingDepth`? What do you think?